### PR TITLE
Some modernization: convert jbuild->dune and update some APIs.

### PR DESCRIPTION
**Pull request by wtipton**
This now compiles against jane bleeding-edge.

### CODE REVIEW

File: src/vg_canvas.ml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 open! Base
 open! Virtual_dom.Vdom
+open Js_of_ocaml
```
Comment (yminsky) [inline src/vg_canvas.ml]: I'd rather not have an open here.  Can we just locally qualify where we use Js_of_ocaml?

File: example/dune [hunk 2]
```diff
@@ -0,0 +1,7 @@
+(executable
+ (name main)
+ (modes js)
+ (libraries incr_dom vg_canvas core_kernel)
+ (preprocess (pps js_of_ocaml-ppx ppx_jane))
+ (flags :standard -short-paths)
```
Comment (yminsky) [inline example/dune]: The -short-paths declaration is no longer needed.

File: dune-project [hunk 3]
```diff
@@ -0,0 +1,5 @@
+(lang dune 2.0)
+
+(name vg_canvas)
+
+(formatting disabled)
```
Comment (yminsky) [inline dune-project]: Is there a reason to disable this?  My instinct would be to just turn it on.

File: dune-project [hunk 4]
```diff
@@ -0,0 +1,5 @@
+(lang dune 2.0)
```
Comment (yminsky) [inline dune-project]: Dune 2.7 maybe? Given that we're up to that now...

**Review by yminsky — COMMENTED** [review]
I left a few small comments, after which, this looks mergable.

File: src/vg_canvas.ml (hunk 1 above)
Comment (wtipton) [reply src/vg_canvas.ml]: Done

File: example/dune (hunk 2 above)
Comment (wtipton) [reply example/dune]: Removed

File: dune-project (hunk 3 above)
Comment (wtipton) [reply dune-project]: Ah I guess the auto-converter put it there for some reason. Removed.

File: dune-project (hunk 4 above)
Comment (wtipton) [reply dune-project]: Changed

**Pull request merged**